upgrade.md

```diff
@@ -0,0 +1,114 @@
+juju integrate <new-hydra-app-name> <new-postgresql-app-name>
+
+juju config postgresql-k8s plugin_uuid_ossp_enable=true
+```
+
+2. Use database migration systems or database replication mechanisms to sync source database to target database. Wait till the target database is synchronized with the source database.
+
+3. Stop writing to the source database by blocking traffic. Wait for all remaining data to drain to the target database. The source and target databases are now fully synchronized.
+
+**IMPORTANT: Make sure to back up your database.**
+
+4. Sanitize the database.
+
+**We recommend performing this step to ensure a seamless upgrade and migration.**
+
+First create a script to clean up the database and transfer it to the postgres charm:
+
+```sql
+-- cleanup-canonical-hydra.sql
+BEGIN;
+
+-- ==========================================================
+-- 1. DROP CANONICAL HYDRA TABLES
+-- ==========================================================
+-- Use 'CASCADE' to remove any foreign key constraints and the tables.
+DROP TABLE IF EXISTS hydra_oauth2_device_code CASCADE;
+DROP TABLE IF EXISTS hydra_oauth2_user_code CASCADE;
+
+-- ==========================================================
+-- 2. CLEANUP 'hydra_oauth2_flow'
+-- ==========================================================
+-- Dropping columns that were added in the Canonical Hydra fork.
+-- Use 'IF EXISTS' to prevent errors if the schema is partially drifted.
+ALTER TABLE hydra_oauth2_flow 
+  DROP COLUMN IF EXISTS device_challenge_id,
+  DROP COLUMN IF EXISTS device_code_request_id,
+  DROP COLUMN IF EXISTS device_verifier,
+  DROP COLUMN IF EXISTS device_csrf,
+  DROP COLUMN IF EXISTS device_user_code_accepted_at,
+  DROP COLUMN IF EXISTS device_was_used,
+  DROP COLUMN IF EXISTS device_handled_at,
+  DROP COLUMN IF EXISTS device_error;
+
+-- ==========================================================
+-- 3. CLEANUP 'hydra_client'
+-- ==========================================================
+ALTER TABLE hydra_client 
+  DROP COLUMN IF EXISTS device_authorization_grant_id_token_lifespan,
+  DROP COLUMN IF EXISTS device_authorization_grant_access_token_lifespan,
+  DROP COLUMN IF EXISTS device_authorization_grant_refresh_token_lifespan;
+
+-- ==========================================================
+-- 4. REMOVE MIGRATION HISTORY
+-- ==========================================================
+-- Delete the version ID that corresponds to the Canonical Hydra
+-- device flow migration. Otherwise, Ory v25.4.0 may
+-- panic because it will see a version ID it doesn't recognize.
+DELETE FROM schema_migration WHERE version = '20240202000001000000';
+
+COMMIT;
+```
+
+```
+juju scp --container postgresql cleanup-canonical-hydra.sql postgresql-k8s/leader:/tmp/cleanup-canonical-hydra.sql
+```
+
+Finally, execute the script against the hydra database.
+
+```sql
+psql --host <psql-host> --username=<your-username> --dbname=<hydra-database-name> -f /tmp/cleanup-canonical-hydra.sql
```

Comment:
Does the username have to be the hydra postgres user?

Reply:
Any user that has an equivalent access to the hydra db. I switched `<your-username>` to `<hydra-user>` so it's more explicit.